ec2/spark_ec2.py

```diff
@@ -750,7 +754,7 @@ def setup_cluster(conn, master_nodes, slave_nodes, opts, deploy_ssh_key):
                'mapreduce', 'spark-standalone', 'tachyon']
 
     if opts.hadoop_major_version == "1":
-        modules = filter(lambda x: x != "mapreduce", modules)
+        modules = list(filter(lambda x: x != "mapreduce", modules))
```

Comment:
Yep.  I was also a bit confused why the list method .remove() isn't just used here.  But I opted for the safer thing which could potentially remove multiple "mapreduce" entries.  Should modules be a set()?